packages/ra-core/src/controller/list/useExpanded.tsx

```diff
@@ -49,3 +49,42 @@ export const useExpanded = (
 
     return [expanded, toggleExpanded];
 };
+
+/**
+ * State-like hook for controlling the expanded state of many list items
+ * expanded state is true when at least one item from ids is expanded.
+ *
+ * @param {string} resource The resource name, e.g. 'posts'
+ * @param {Identifier[]} ids A list of record identifiers
+ * @returns {Object} Destructure as [expanded, toggleExpanded].
+ *
+ * @example
+ *
+ * const [expanded, toggleExpanded] = useExpandedMultiple('posts', [123, 124, 125]);
+ * const expandIcon = expanded ? ExpandLess : ExpandMore;
+ * const onExpandClick = () => toggleExpanded();
+ */
+export const useExpandedMultiple = (
+    resource: string,
+    ids: Identifier[]
+): [boolean, () => void] => {
+    const [expandedIds, setExpandedIds] = useStore<Identifier[]>(
+        `${resource}.datagrid.expanded`,
+        []
+    );
+
+    const expanded = Array.isArray(expandedIds)
```

Comment:
This variable name is still not explicit enough IMO. We already have an `expandedIds`, so this `expanded` is confusing. 

Why not rename `expandedIds` to `allExpandedId`, and `expanded` into `displayedExpandedIds`, or something like that?

Reply:
this variable name is literally as explicit as in useExpanded.
I also don't see a reason to rename `expandedIds` just because the context is different.

Example:

``` typescript
function square(x: number) {
    const exponent = 2;
    return Math.pow(x, exponent);
}

function squareAll(x: number[]) {
    const exponent = 2;
    return x.map(value => Math.pow(value, exponent));
}
```

would you rename exponent to exponentAll?

Reply:
You named this variable `expanded`, but it's *a filtered list of expanded ids*. The name you chose is misleading and not precise enough.

This may look like nitpicking, but we'll have to maintain this code for years. We have to be confident we understand it well.

Reply:
Sorry, I re-read your code and it seems this variable contains a boolean, not a selection of ids. To make it explicit, please name it `isExpanded`, or better, `hasSomeExpanded`

Reply:
Don't get me wrong. It's your project so you decide.

I just wasn't very happy with your suggestion and since we mimic a similar behavior as in useExpanded, my personal take is to use a similar naming convention.
I went for isExpanded now, since hasSomeExpanded gives more confusion in the toggleExpandedAll function